### PR TITLE
Add Model Registry scaffolding

### PR DESCRIPTION
**Pull request by lucferbux**
Closes https://issues.redhat.com/browse/RHOAIENG-2112

## Description

Add Model Registry setup configuration.

Screenshot 2024-03-22 at 02 37 21

## How Has This Been Tested?

1. In a new cluster, deploy the new OdhDashboardConfig crds and crs
2. Open the dashboard, model registry should be disabled
3. Change `disableModelRegistry` attribute in `OdhDashboardConfig` to `false`
4. Check that the section is now enabled

## Test Impact

Added first tests to check area

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/areas/types.ts [hunk 1]
```diff
@@ -58,6 +61,7 @@ export enum StackComponent {
   WORKBENCHES = 'workbenches',
   TRUSTY_AI = 'trustyai',
   KUEUE = 'kueue',
+  MODEL_REGISTRY = 'trustyai', // TODO: Change to model-registry once the component is available in DSC
```
Comment (lucferbux) [inline frontend/src/concepts/areas/types.ts:64]: This is just a workaround in place, if not it's gonna be really hard to test the feature until is enabled by the operator.

cc @zdtsw any idea on how to hack a cluster to show an unavailable dsc component?

File: frontend/src/concepts/areas/const.ts [hunk 2]
```diff
@@ -66,4 +66,8 @@ export const SupportedAreasStateMap: SupportedAreasState = {
     featureFlags: ['disableDistributedWorkloads'],
     requiredComponents: [StackComponent.KUEUE],
   },
+  [SupportedArea.MODEL_REGISTRY]: {
+    featureFlags: ['disableModelRegistry'],
+    requiredComponents: [StackComponent.TRUSTY_AI], // TODO: Change to model-registry once the component is available in DSC
```
Comment (lucferbux) [inline frontend/src/concepts/areas/const.ts:71]: This is just a workaround in place, if not it's gonna be really hard to test the feature until is enabled by the operator.

cc @zdtsw any idea on how to hack a cluster to show an unavailable dsc component?

**Review by dpanshug — APPROVED** [review]
/lgtm

File: frontend/src/concepts/areas/const.ts [hunk 3]
```diff
@@ -66,4 +66,8 @@ export const SupportedAreasStateMap: SupportedAreasState = {
     featureFlags: ['disableDistributedWorkloads'],
     requiredComponents: [StackComponent.KUEUE],
```
Comment (zdtsw) [inline frontend/src/concepts/areas/const.ts:67]: this is intetsting, DW is based on component kueue, what happens if Ray or CFO are set to different managementstate than kueue? and soon coming TO
@lucferbux / @anishasthana

Comment (lucferbux) [reply frontend/src/concepts/areas/const.ts:67]: This is a great question cc @gitdallas @mturley

**Review by dpanshug — APPROVED** [review]
/lgtm

**Pull request merged**